source/utilities.tex

```diff
@@ -14759,6 +14826,150 @@
+\item
+Otherwise, if \placeholder{C} is not \unicode{0020}{space} and
+
+\begin{itemize}
+\item
+\placeholder{CE} is a Unicode encoding and
+\placeholder{C} corresponds to either
+a UCS scalar value whose Unicode property \tcode{General_Category}
+has a value in the groups \tcode{Separator (Z)} or \tcode{Other (C)} or to
+a UCS scalar value which has the Unicode property \tcode{Grapheme_Extend=Yes},
+as described by table 12 of UAX \#44, or
+
+\item
+\placeholder{CE} is not a Unicode encoding and
+\placeholder{C} is one of an implementation-defined set
+of separator or non-printable characters
+\end{itemize}
+
+then the sequence \tcode{$\backslash$u\{\placeholder{hex-digit-sequence}\}}
+is appended to \placeholder{E},
+where \placeholder{hex-digit-sequence}
+is the shortest hexadecimal representation
+of \placeholder{C} using lower-case hexadecimal digits.
+
+\item
+Otherwise, \placeholder{C} is appended to \placeholder{E}.
+\end{itemize}
+
+\item
+Otherwise, if \placeholder{X} is a shift sequence,
+the effect on \placeholder{E} and further decoding of \placeholder{S}
+is unspecified.
+It is recommended that
+a shift sequence be represented in \placeholder{E}
+such that the original code unit sequence of \placeholder{S}
+can be reconstructed.
+
+\item
+Otherwise (\placeholder{X} is a sequence of ill-formed code units),
+each code unit \placeholder{U} is appended to \placeholder{E} in order
+as the sequence \tcode{$\backslash$x\{\placeholder{hex-digit-sequence}\}},
+where \placeholder{hex-digit-sequence}
+is the shortest hexadecimal representation of \placeholder{U}
+using lower-case hexadecimal digits.
+\end{itemize}
+
+\item
+Finally, \unicode{0022}{quotation mark} (\tcode{"})
+is appended to \placeholder{E}.
+\end{itemize}
+%
+\begin{floattable}{Mapping of characters to escape sequences}{format.escape.sequences}{ll}
+\topline
+\lhdr{Character} & \rhdr{Escape sequence} \\ \rowsep
+\unicode{0009}{character tabulation} &
+\tcode{$\backslash$t}
+\\ \rowsep
+%
+\unicode{000a}{line feed} &
+\tcode{$\backslash$n}
+\\ \rowsep
+%
+\unicode{000d}{carriage return} &
+\tcode{$\backslash$r}
+\\ \rowsep
+%
+\unicode{0022}{quotation mark} &
+\tcode{$\backslash$"}
+\\ \rowsep
+%
+\unicode{005c}{reverse solidus} &
+\tcode{$\backslash\backslash$}
+\\
+\end{floattable}
+
+\pnum
+The escaped string representation of a character \placeholder{C}
+is equivalent to the escaped string representation
+of a string of \placeholder{C}, except that:
+
+\begin{itemize}
+\item
+the result starts and ends with \unicode{0027}{apostrope} (\tcode{'})
+instead of \unicode{0022}{quotation mark} (\tcode{"}), and
+\item
+if \placeholder{C} is \unicode{0027}{apostrope},
+the two characters \tcode{$\backslash$'} are appended to \placeholder{E}, and
+\item
+if \placeholder{C} is \unicode{0022}{quotation mark},
+then \placeholder{C} is appended unchanged.
+\end{itemize}
+
+%% FIXME: Example is incomplete; s2 and s6 are missing below;
+%% FIXME: their Unicode characters break the build.
```

Comment:
@tkoeppe @jensmaurer, How to resolve this?

Reply:
All Greek to me - I'd prefer to leave this to the experts, but if you have some specific changes in mind I'll give them a try.

Reply:
Actually the simplest solution would be to just replace the emojis with accented or non-Latin characters that require more than one UTF-8 code unit. Those might be more likely to exist in the fonts we use.

Reply:
No, the example is specifically about graphemes clusterization, so we cannot use arbitrary text, and we didn't want to mix normalization and clusterization to avoid confusion. There are certainly other options, ie other emojis (which have the benefits of being generally familiar to most people, at least on some level), or katakanas possibly. 

I can try to come up with alternatives.

Reply:
I made a pull-request here https://github.com/cplusplus/draft/pull/5702
I was not able to reproduce the emoji example, that would require switching to luatex, so I used another grapheme extending example.

![Screenshot_20220804_132040](https://user-images.githubusercontent.com/1409019/182834954-60649ac4-f8e4-4a29-b4a8-7bf5c62f1db8.png)

Reply:
To clarify, all of this is non-normative, and up to editorial changes, right?

Reply:
Absolutely. We should not hold merging this for the examples.
You could keep the s6 example for now as there is no unicode characters involved, if you want.

I wanted to keep s2 as is as the fact it spells "Thanks Victor" is a nice touch. But, it's not important.
I also played with porting the standard to luatex to see if we could support unicode everywhere more easily, i wonder what you think about that. but again, it's all non normative.

Reply:
We'll not change the LaTeX engine or setup right now. We can revisit that in the future.

For now, a simple solution could be to render a few examples from `standalone` xetex sources as PDFs, and we include them here with `\includegraphics`. That should end up embedding the fonts nicely in the final document.

The remaining question is which fonts to use that a) keep everything legal and b) harmonize with Latin Modern. Suggestions welcome!